examples/charm++/future/Makefile

```
@@ -0,0 +1,30 @@
+-include ../../common.mk
+CHARMC = ../../../bin/charmc
+BINARY = fib
+
+CHARMCFLAGS = $(OPTS)
+CHARMCLINKFLAGS = -language charm++ $(OPTS)
+
+%.o: %.C
+
+all: $(BINARY)
+$(BINARY): $(patsubst %.C,%.o,$(wildcard *.C))
+	$(CHARMC) $(CHARMCLINKFLAGS) -o $@ $+ 
+
+.SECONDARY: $(patsubst %.C,%.decl.h,$(wildcard *.C))
+.SECONDARY: $(patsubst %.C,%.def.h,$(wildcard *.C))
+
+%.o: %.C %.decl.h %.def.h
+	$(CHARMC) $(CHARMCFLAGS) $<
+
+%.decl.h %.def.h: %.ci
+	$(CHARMC) $(CHARMCFLAGS) $<
```

This Makefile construction for building decl/def is unsafe because it will run charmc on the .ci file _twice_, once for each of the two files, and there will sometimes be a race condition in each charmxi process' file writing which results in malformed header output. I recommend defining the build rule only for the decl, and adding a new build rule for the def that is chained to the decl but does nothing of its own.

```suggestion
%.decl.h: %.ci
	$(CHARMC) $(CHARMCFLAGS) $<

%.def.h: %.decl.h
```

---

Thanks! This was copied verbatim from `examples/charm++/fib` so we may want to backport our preferred fix.